pkg/reconciler/certificate/certificate.go

```diff
@@ -101,26 +101,20 @@ func (c *Reconciler) reconcile(ctx context.Context, knCert *v1alpha1.Certificate
 
 	cmConfig := config.FromContext(ctx).CertManager
 
-	cmCert, err := resources.MakeCertManagerCertificate(cmConfig, knCert)
-	if err != nil {
-		return err
+	cmCert, errCondition := resources.MakeCertManagerCertificate(cmConfig, knCert)
+	if errCondition != nil {
+		certificateCondSet.Manage(&knCert.Status).SetCondition(*errCondition)
+		knCert.Status.MarkNotReady(errCondition.Reason, errCondition.Message)
```

Comment:
Looks like `MarkNotReady` is really setting the status to `Unknown` which is wrong.

What we want is `MarkFailed` to make `Ready=False`

https://github.com/knative/networking/blob/c692e9e6afe17fdd870e19fd04a869585296e315/pkg/apis/networking/v1alpha1/certificate_lifecycle.go#L42

Reply:
Regarding this, and your previous comment about the status. I did it this way because the reconcile just sets a blanket `Ready=Unknown` on any reconcile error: https://github.com/KauzClay/net-certmanager/blob/ck-issue-214-followup/pkg/reconciler/certificate/certificate.go#L88

I was hesitant to change too much of that behavior, but I can